### PR TITLE
Improve conceptual integrity of SVM nomenclature 

### PR DESCRIPTION
**Pull request by dsouzai**
The SVM, at its core, maintains a mapping between the results of front end VM queries and `uint16_t` IDs. Conceptually, the ID is the symbol that is associated with the result which is the value. However, in the SVM, the term "symbol" was used to refer to the value. This is confusing for anyone new to the SVM, especially since it claims to be similar to a symbol table in a linker.

This PR improves the conceptual integrity of the SVM by actually referring to values as values and IDs as symbols.

### CODE REVIEW

**Comment by dsouzai** [comment]
@jdmpapin could you please review?

**Comment by dsouzai** [comment]
@jdmpapin review reminder for this.

File: doc/compiler/aot/SymbolValidationManager.md [hunk 1]
```diff
@@ -58,18 +58,18 @@ guaranteed to be compatible in the current environment.
...
-associate unique IDs to each new symbol the compiler gets a hold of. Thus,
-when a symbol is acquired by asking a query using a different symbol, the
-validation record that gets generated refers to these two symbols by their
-IDs, which are environment agnostic. In the load run, the AOT load
-infrastructure uses these IDs to build up its table of symbols that are
+associate unique IDs to each new value the compiler gets a hold of. Thus,
+when a value is acquired by asking a query using a different value, the
+validation record that gets generated refers to these two values by their
+IDs (symbols), which are environment agnostic. In the load run, the AOT load
+infrastructure uses these IDs to build up its table of value that are
```
Comment (jdmpapin) [inline doc/compiler/aot/SymbolValidationManager.md:65]: This should be "it's table of _values_ [plural] that are"

**Review by jdmpapin — APPROVED** [review]
LGTM with a few very minor comments

There's a typo in the commit message of b9f31905 where it says `definedGuaranteedID` (past/adjective) instead of `defineGuaranteedID` (imperative)

File: runtime/compiler/control/JITClientCompilationThread.cpp [hunk 2]
```diff
@@ -3476,7 +3476,7 @@ remoteCompile(J9VMThread *vmThread, TR::Compilation *compiler, TR_ResolvedMethod
             // so need to exit heuristic region.
             compiler->exitHeuristicRegion();
             // Populate symbol to id map
-            compiler->getSymbolValidationManager()->deserializeSymbolToIDMap(svmSymbolToIdStr);
+            compiler->getSymbolValidationManager()->deserializeValueToSymbolMap(svmSymbolToIdStr);
```
[the remark this answers is not in the document]
Comment (dsouzai) [reply runtime/compiler/control/JITClientCompilationThread.cpp]: Updated in force push https://github.com/eclipse-openj9/openj9/compare/bf8608af47dbd1df69cf073d45cf70d166e29a25..f23a15e3597cc6e11fdaf9712f42e58e6c9ebcff

File: doc/compiler/aot/SymbolValidationManager.md (hunk 1 above)
Comment (dsouzai) [reply doc/compiler/aot/SymbolValidationManager.md]: Updated in force push https://github.com/eclipse-openj9/openj9/compare/bf8608af47dbd1df69cf073d45cf70d166e29a25..f23a15e3597cc6e11fdaf9712f42e58e6c9ebcff

**Comment by dsouzai** [comment]
@jdmpapin made the requested changes (including the commit message typo). Good for final review now.

**Review by jdmpapin — APPROVED** [review]

**Comment by dsouzai** [comment]
Jenkins test sanity+aot all jdk17

**Comment by jdmpapin** [comment]
```
/home/jenkins/workspace/Build_JDK17_aarch64_linux_Personal/openj9/runtime/compiler/codegen/J9AheadOfTimeCompile.cpp:1290:61: error: 'class TR::SymbolValidationManager' has no member named 'getIDFromSymbol'; did you mean 'getValueFromSymbolID'?
          thunkRecord->setThunkID(reloTarget, symValManager->getIDFromSymbol(svmRecord->_thunk));
                                                             ^~~~~~~~~~~~~~~
                                                             getValueFromSymbolID
/home/jenkins/workspace/Build_JDK17_aarch64_linux_Personal/openj9/runtime/compiler/codegen/J9AheadOfTimeCompile.cpp:1291:62: error: 'class TR::SymbolValidationManager' has no member named 'getIDFromSymbol'; did you mean 'getValueFromSymbolID'?
          thunkRecord->setMethodID(reloTarget, symValManager->getIDFromSymbol(svmRecord->_method));
                                                              ^~~~~~~~~~~~~~~
                                                              getValueFromSymbolID
```
Looks like you'll need to rebase and update these lines from #14542

**Comment by jdmpapin** [comment]
Jenkins test sanity+aot all jdk17

**Comment by dsouzai** [comment]
jenkins woes again :(

**Comment by dsouzai** [comment]
Jenkins test sanity+aot all jdk17

**Comment by jdmpapin** [comment]
Jenkins test sanity+aot all jdk17

**Comment by AdamBrousseau** [comment]
Jenkins test sanity.functional+aot all jdk17

**Comment by dsouzai** [comment]
AIX build failed because of 
```
[2022-03-05T02:28:56.690Z]  [ERR] Exception in thread "main" java.awt.AWTError: Can't connect to X11 window server using 'unix:0' as the value of the DISPLAY variable.
[2022-03-05T02:28:56.690Z]  [ERR] 	at java.desktop/sun.awt.X11GraphicsEnvironment.initDisplay(Native Method)
[2022-03-05T02:28:56.690Z]  [ERR] 	at java.desktop/sun.awt.X11GraphicsEnvironment$1.run(X11GraphicsEnvironment.java:101)
...
```
which, even though it happens on the second run, has been seen before outside of this PR: https://github.com/eclipse-openj9/openj9/pull/14421#issuecomment-1035067528

@jdmpapin This PR should be good for merging now.

**Pull request merged**